### DISCUSSION TITLE
Connection termination response code 502 error

### DISCUSSION THREAD

**Discussion by parshva-shah-dsw** [istio/istio#48935] [General]
in the backend application error log is printed with upstream reset before response started {protocol_error}

**Comment by zirain** [comment]
please follow https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/ and try again.

**Reply by parshva-shah-dsw** [reply]
@zirain we are observing it for only one api and the related application pod. If it would be an protocol error all the flows should have been failing. application calls are done with http protocol only
current k8s service definition 
port: 3200
protocol: tcp
targetPort: 3200
